src/api/v4/dataview/histogram.js

```diff
@@ -0,0 +1,186 @@
+ * 
+ * @param {number} bins
+ * @return {carto.dataview.Histogram} this
+ * @api
+ */
+Histogram.prototype.setBins = function (bins) {
+  this._validateBins(bins);
+  var floorBins = Math.floor(bins);
+  this._changeProperty('bins', floorBins);
+  return this;
+};
+
+/**
+ * Return the current number of bins
+ * 
+ * @return {number} Current number of bins
+ * @api 
+ */
+Histogram.prototype.getBins = function () {
+  return this._bins;
+};
+
+/**
+ * Return the distribution type of the current data according to Galtung’s AJUS System http://druedin.com/2012/12/08/galtungs-ajus-system/
+ * 
+ * @return {string} Distribution type of current data
+ * @api 
+ */
+Histogram.prototype.getDistributionType = function () {
+  if (this._internalModel) {
+    var data = this._internalModel.getData();
+    return this._internalModel.getDistributionType(data);
+  }
+  return null;
+};
+
+/**
+ * Return the distribution type of the totals data according to Galtung’s AJUS System http://druedin.com/2012/12/08/galtungs-ajus-system/
+ * 
+ * @return {string} Distribution type of current data
+ * @api 
+ */
+Histogram.prototype.getTotalsDistributionType = function () {
+  if (this._internalModel) {
+    var data = this._internalModel.getUnfilteredData();
+    return this._internalModel.getDistributionType(data);
+  }
+  return null;
+};
+
+Histogram.prototype._validateBins = function (bins) {
+  if (!_.isFinite(bins) || bins < 1) {
+    throw new TypeError('Bins must be a positive value.');
+  }
+};
+
+Histogram.prototype._parseData = function (data, nulls, totalAmount) {
+  if (data) {
+    /**
+     * @typedef {object} BinItem
+     * @property {number} index
+     * @property {number} min - Only appears if freq > 0
+     * @property {number} max - Only appears if freq > 0
+     * @property {number} avg - Only appears if freq > 0
+     * @property {number} freq
+     * @property {number} normalized
+     * @api
+     */
+    /**
+     * @typedef {object} HistogramData
+     * @property {number} nulls
+     * @property {number} totalAmount
+     * @property {BinItem[]} result
+     * @property {string} type - Constant 'histogram'
+     * @api
+     */
+    var maxFreq = _.max(data, function (bin) {
+      return bin.freq || 0;
+    }).freq;
+    var result = data.map(function (bin) {
+      return _.extend(bin, {
+        normalized: _.isFinite(bin.freq)
+          ? bin.freq / maxFreq
+          : 0
+      });
+    });
+
+    return {
+      result: result,
+      nulls: nulls || 0,
+      totalAmount: totalAmount
+    };
+  }
+  return null;
+};
+
+Histogram.prototype._checkOptions = function (options) {
+  if (_.isUndefined(options)) {
+    throw new TypeError('Options object to create a histogram dataview is required.');
+  }
+  this._validateBins(options.bins);
+  options.bins = Math.floor(options.bins);
+};
+
+Histogram.prototype._listenToInternalModelSpecificEvents = function () {
+  this.listenTo(this._internalModel, 'change:bins', this._onBinsChanged);
```

Comment:
Just being paranoid. We're providing events to notify in the wrapper objects that the dataviews properties have changed. Usually, it'll be the API user the one who'll change it. 
This code is to prevent something changing the internal model and having the wrapped object and the internal one not synchronised.